### PR TITLE
Use correct `cc_engine_instrumentation` in `jazzer` config

### PR DESCRIPTION
**Pull request by fmeum**
While the `cc_engine_instrumentation` settings `libfuzzer` and `jazzer` lead to the same instrumentation, this may not remain so indefinitely. Thus the `jazzer` and `asan-jazzer` config in `.bazelrc` should use the `jazzer` value for the setting.

Also, even when not using a sanitizer via the `jazzer` config, native library dependencies should be instrumented for fuzzing.

This requires updating Jazzer to the latest version as previous versions did not correctly export certain sanitizer symbols.

### CODE REVIEW

**Comment by fmeum** [comment]
Not sure why this issue has only appeared now, but it will be resolved by the extended logic for populating `LD_LIBRARY_PATH` in #157. This one can wait until the PR has been merged.

**Comment by stefanbucur** [comment]
> Not sure why this issue has only appeared now, but it will be resolved by the extended logic for populating `LD_LIBRARY_PATH` in #157. This one can wait until the PR has been merged.

This PR looks good to me, will approve and merge once #157 lands.

**Comment by fmeum** [comment]
The checks pass now.

**Review by stefanbucur — APPROVED** [review]

**Pull request merged**
